### ISSUE TITLE
Create a daemon from SlipStreamClient with an API instead of using CLIs

### ISSUE THREAD

**Issue by schaubl** [slipstream/SlipStream#126]
Currently (for historical reasons) when the SlipStream server has to do an action on a Cloud it will start a Python CLI (fork of a new Python interpreter).

This is costly especially considering the fact that a lot of actions are done on Clouds even for a medium size SlipStream deployment.

There is currently only 3 available Clod actions:
- List instances
- Start an instance
- Terminate instances

To add more actions it will require to create more CLIs which is not convenient.

The suggestion here is to create a service (daemon) with the SlipStreamClient to accept requests from the SlipStream server and then forward them to the right connector.
The API could be a simple [JSON-RPC](http://www.jsonrpc.org/) API.

An extension to that would be to use an [AMQP](https://en.wikipedia.org/wiki/Advanced_Message_Queuing_Protocol) server (like [rabbitmq](https://www.rabbitmq.com/)) to queue and distribute API calls.
With this it's easy to scale horizontally by simply adding more daemons.

The SlipStreamServer will have to updated to do API calls instead of CLI calls.

To summarise this will bring the following benefits:
- Make SlipStream easier to extend.
- Highly reduce the load of the system (no fork, no loading of the Python interpreter everytime, no need to load the connector everytime)
- Reduce the RTT by keeping sessions (tokens, cookie, ...) and connection (keep-alive).
- Can be scaled horizontally with no effort.

**Comment by loomis** [comment]
Needs discussion at a "big picture" meeting to define full feature.

**Comment by schaubl** [comment]
This has been implemented with the JobEngine.
Currently only the `list instances` go through the job engine.
